sound: create sandbox example for text to speech library pull request

JaStarbuck opened the pull request:
Summary: We have successfully cloned the espeak repo into the VM and followed the demo to create the test-espeak.c file in the sandbox folder of the sound directory. After getting driver warnings from computers that don't have speakers, we were able to get it to run on a pc that has speakers completely without errors. In the future, we will hopefully be able to generalize the process in the demo to create functions to use text to speech during a chiventure game.

first comment: Added the test-espeak.c file from https://github.com/espeak-ng/espeak-ng/blob/master/docs/integration.md and cloned the espeak repo so that running ./test-espeak will output sound from the VBox. There is an accidental minor change to sound.c that can be ignored.

JaStarbuck committed:
progress 1

JaStarbuck committed:
got it working!

JaStarbuck committed:
removed accidental changes to sound.c

JaStarbuck committed:
added some comments

jiaoboxuan reviewed (approved):
Looks good. That said, the comment looks like it belongs to a `README.md` file perfectly, so I suggest you do so. Especially considering it's a sandbox, future groups could pick them up easily if they can locate a `README.md`.

ecoble committed:
Merge branch 'dev' into sound/tts_jason

ecoble commented on src/sound/sandbox/test-espeak.c:
```diff
@@ -0,0 +1,25 @@
+#include <espeak-ng/speak_lib.h>
+ 
+// in order to run this, in the chiventure/src/sound/sandbox directory run 
```
Please make this a block comment like this 
/*
 *
 */

ecoble commented on src/sound/sandbox/test-espeak.c:
```diff
@@ -0,0 +1,25 @@
+
+ espeak_AUDIO_OUTPUT output = AUDIO_OUTPUT_SYNCH_PLAYBACK;
+ char *path = NULL;
+ void* user_data;
```
Please switch asterisk to the other side for consistent style, you can fix style automatically using astyle, more documentation [here](https://uchicago-cs.github.io/dev-guide/style_guide_c.html#style-guide-c).

ecoble commented on src/sound/sandbox/test-espeak.c:
```diff
@@ -0,0 +1,25 @@
+ void* user_data;
+ unsigned int *identifier;
+ 
+ int main(int argc, char* argv[]) {
```
Asterisk style

ecoble reviewed (changes requested):
This looks good besides some style issues! You can leave the comment in this file for now, but a README containing this as well as installation instructions is a good idea.

JaStarbuck committed:
Merge branch 'dev' into sound/tts_jason

JaStarbuck committed:
fixed style issues

JaStarbuck committed:
I need to pull so that I can edit and fix style changes
Merge branch 'sound/tts_jason' of https://github.com/uchicago-cs/chiventure into sound/tts_jason

JaStarbuck committed:
used all style

jiaoboxuan reviewed (approved):
Looks about okay.

ecoble reviewed (approved):
Nice work! Excited to see the demo!

jwschroeder21 commented:
closing statement: 

We first successfully cloned the espeak repo into the VM.  We then followed the demo to create the test-espeak.c file in the sandbox folder of the sound directory. Then, after getting driver warnings from computers that don't have speakers, we ran it on a PC that had speakers (no errors!). 

Our goal is to generalize the process in the demo to create functions to use text to speech during a chiventure game.

With Elizabeth and Victor's approval, this PR has been successfully merged!

ecoble commented:
Issue Score: Excellent

Comments:
Nice job with e-speak, integrating a new library can be really challenging and you did a great job!

<!-- {"score": "excellent", "minor": false} -->
